### ISSUE TITLE
Redirect Request: Vet Center QR code URL - ASAP target timing

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#11662]
**BLOCKED on WebOps: for subdomain > subdomain redirect, our attempted code change did not work. We will need WebOps to perform this redirect. WebOps ticket opened, TBD response.**

### Type of request
- [X] We are changing the URL of an existing page (complete redirect section)

## Context
From Tiffany Lever via email to DaveC: 
> Recently, we came across a QR code that directs people to “Vet Centers are Here for you 24/7”. Since the QR code could continue to be out there and used, would it be possible to have https://www.vetcenter.va.gov/Vet_Centers_are_Here_For_You_24_7.asp be redirected to https://www.vetcenter.va.gov/? Thank you, let me know if there is anything I can do to help make this happen.  Happy Friday and have a great day!

### Implementation date
When does this request need to be live: ~~no hard timing~~ ASAP since fix for incorrect link in the wild

### Redirects 
Current URL  |  Redirect Destination or New URL
---  |  ---
https://www.vetcenter.va.gov/Vet_Centers_are_Here_For_You_24_7.asp | https://www.vetcenter.va.gov/

## FE Implementation notes
This will need to be done via [client side redirect](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#4-client-side-redirects-for-subdomains-subdomainvagov). 
* We have confirmed that proxy-rewrite JS is being executed on vetcenter.va.gov. 
* Please follow documentation for how to implement via [client side redirects](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#4-client-side-redirects-for-subdomains-subdomainvagov).

### Process, Roles and Responsibilities
- [ ] Requesting team: Above information is provided
- [ ] Requesting team: All appropriate VA stakeholders are notified as appropriate
- [ ] IA: Request is vetted and documented and implementation plan is clear 
- [ ] IA: Request is assigned to appropriate team for implementation 
- [ ] Implementation team: Work is complete
- [ ] Implementation team: Validated in production
- [ ] Requesting team: Validates everything is correct in production and closes ticket

**Comment by chri5tia** [comment]
PR: https://github.com/department-of-veterans-affairs/devops/pull/12277

**Comment by jilladams** [comment]
Learned stuff with Dave's help: 

- [Redirect Implementation strategy](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md) (last update Jan 2020)
  - Lists [subdomains](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#examples-of-redirects---current-approach) where we can implement server-side redirects, and [vetcenter.va.gov](http://vetcenter.va.gov/) is not included in the list
  - [How subdomain redirects work](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#3-server-side-redirects-from-subdomains-subdomainvagov), including 2 notes: 
     - "Option 3(?): Can a server-side redirect be implemented in Team Site?"
    - "For the majority of page-level redirects from subdomains ([subdomain.va.gov](http://subdomain.va.gov/)), this is not feasible in the short term because it requires access to and relationships with technical stakeholders on the relevant teams. These contacts should be actively sought out via DEPO."

From [va.gov-team/Web_Modernization_Run_Book.md at master · department-of-veterans-affairs/va.gov-team (github.com)](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/teams/vamc/vamc_transition_documents/Web_Modernization_Run_Book.md)
> In cases where you are redirecting multiple pages considered more of a full website of pages, you will need to redirect at the server level.  To accomplish this, you will need to engage the Web Operations team and have full authorization from the authorized owner of the website (e.g., VHA Digital Media).  These considerations and steps are outlined below.


Thusly: have opened a ServiceNow ticket for WebOps help: https://yourit.va.gov/va?id=ticket&is_new_order=true&table=incident&sys_id=47f15a951bef555018da5316624bcbe1

(And: [how to do that](https://dsva.slack.com/archives/C52CL1PKQ/p1670023933435989), if needed in future)

**Comment by jilladams** [comment]
Unassigning / removed from sprint, since this became a VA admin thing and not a PW code thing.

**Comment by jilladams** [comment]
We've opened a SNOW ticket

**Comment by jilladams** [comment]
Updates: 

1, I apparently didn't understand how to send a SNOW ticket to WebOps and have been asked to reassign it to WebOps. Figuring that out. 

2, Meanwhile: Ryan and I re-reviewed the info around [client side redirects](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#4-client-side-redirects-for-subdomains-subdomainvagov). Some takeaways: 

Client-side redirect is only possible if the proxy-rewrite JS is being executed on the relevant subdomain. In the case of `vetcenter.va.gov`, this subdomain has been added to support the injected header ([here](https://github.com/department-of-veterans-affairs/vets-website/blame/66479197a3d5eebdbb2e4806b925b00a468c5116/src/applications/proxy-rewrite/proxy-rewrite-whitelist.json#L165) via what looks like a [proactive effort](https://github.com/department-of-veterans-affairs/vets-website/pull/12438) to get our header on high priority subdomains). The header is in testing mode, visible on vetcenter.va.gov only when cookie is enabled. 

This means: 

1. Our proxy-rewrite JS is executing on the subdomain
2. We can do client-side redirect if that's the best thing to do. Caveat is that docs flag: 
    * This is not ideal, as it depends on our JavaScript running correctly in order for a user to be redirected properly. It also depends on the legacy page staying live. Should be considered temporary solution.
    * If a page gets archived that has an existing client-side redirect, users may first see a 404 prior to the redirect executing. So if this page being redirected ever needs to be archived, that should actually be done via A) Replacing their page content with a 'redirect' message and B) Update their print materials and nav links with the new URL
4. Future work: we can / maybe should follow up with vetcenter subdomain owners about what it would take to turn on the injected header. Ticket cut for that:  #11853 11853

@ryguyk could you update points here for your best guess, assuming we go the client-side redirect route?

**Comment by jilladams** [comment]
Merged, leaving open to get verified in prod after deploy.

**Comment by jilladams** [comment]
Sad news, friends @jtmst @ryguyk : This redirect isn't working. 

Verified that code is live in prod: 
![Screen Shot 2022-12-20 at 2 09 56 PM](https://user-images.githubusercontent.com/85581471/208776428-30641ff0-fbde-45e4-a8d1-f2003a84f444.png)

https://www.vetcenter.va.gov/Vet_Centers_are_Here_For_You_24_7.asp still lands / does not forward to the new URL. 

Moving ticket back to Stretch, and we'll need to work out new approach. I'm gonna start by trying to figure out how to properly request something from WebOps via ServiceNow.

**Comment by jilladams** [comment]
Found a portal to make requests direct to WebOps instead of through ServiceNow: https://vaww.webops.va.gov/info/?get=catalog

Filed a redirect request there today. Will see what sort of response we get -- guessing prob not much in this holiday week, but tbd. Have also closed loop with stakeholder who made this request, and is 2ndary contact on the WebOps ticket. 

Not sure if it's worth doing much more on our side here until we hear from them.

**Comment by jilladams** [comment]
Note: I can't find any artifact of the WebOps ticket. 🤷‍♀️ Didn't receive an email, and the WebOps ticketing UI did confirm successful submission, but I'm not finding any way to see my open issues. So: unclear if/when we may get traction on that.

**Comment by jilladams** [comment]
No news from WebOps ticket.

**Comment by jilladams** [comment]
New attempt: 

Following steps from https://vaww.webops.va.gov/apps/kbx/kbarticle.cfm?get=2018-CST-0416041924, filed a new YourIT ticket to WebOps: 

https://yourit.va.gov/va?id=ticket&is_new_order=true&table=incident&sys_id=00346bb71b806d10da76a798624bcb07

**Comment by jilladams** [comment]
The plot thickens: 
> WebOps only create site level redirects. Page level redirects are the site editor's responsibility. 
> This is a TeamSite website and is supported by VA Web CMS Support <VAWebCMSSupport@va.gov>. Transferring to Sandeep Kotian/VA Web Solutions support for assistance.

The issue was reassigned to Sandeep, our TeamSite friend.
Sandeep then referenced an index of site editors and asked Jenny Heiland-Luedtke. Deputy Director of VHA Digital Media, if someone within her group could manage this redirect, which they cannot. 

Per discussion with Jenny: 
* Confirmed that WebOps will only handle full site redirects, not one-off URL redirects
* She thought she recalled that the previous Public Websites team _was_ able to do redirects to and from a subdomain. So we need to revisit what we know and understand exactly why we don't think we can execute on this. May be some disconnect with how we are understanding the previous team's documentation.

**Comment by jilladams** [comment]
After auditing existing docs, I don't find any indication that the PW team can directly implement a page-level redirect between 2 subdomain URLs. (In this case: both to/from are on the vetcenter.va.gov subdomain, powered by TeamSite, that _does_ run our proxy-rewrite js but does _not_ currently display the injected header.) These docs seem to indicate that:
1) page level redirects from a subdomain to va.gov must be done client side, if at all; 
2) page level redirects _within_ a subdomain weren't really considered or documented
3) there was some work remaining identified, to better understand how server-side redirects might be achieved for subdomains, which might solve this use case as well. 

I'd feel better about that assessment if someone with a better working knowledge of reverse proxies and TICs reviewed docs #1 and #2 here to see if they agree. 

And if so: my suggestion would be for me to email Sandeep Kotian, Brad Smelley, and Jenny H-L on the existing email thread with Tiffany Lever / Dave / Wes to figure out how TeamSite folks can / can't help us.

# Existing redirect documentation
## 1. [Redirects](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/teams/vsa/teams/public-websites/redirects.md)  
`va.gov-team-sensitive/teams/vsa/teams/public-websites/redirects.md`
_July 2021_

Describes 3 types of redirect: 
1. **Same-domain redirect**: from va.gov to www.va.gov — via revproxy-vagov [redirects.yml](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/deployment/config/revproxy-vagov/vars/vetsgov-redirects.yml) file changes
2. **vets.gov cross-domain redirect** — from www.vets.gov to www.va.gov — via revproxy-vagov redirects.yml file changes
3. **cross-domain redirect** — from www.[subdomain].va.gov to va.gov — has 3 flavors: 
  a. TeamSites which do _not_ use the injected header — TeamSite folks must handle the redirect.
  b. TeamSites which use the injected header — redirect via [vets-website/src/applications/proxy-rewrite/redirects/crossDomainRedirects.json](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/proxy-rewrite/redirects/crossDomainRedirects.json).   (Which we tried via [PR 22944](https://github.com/department-of-veterans-affairs/vets-website/pull/22944/files), and did not work.)
  c. Other sites that do _not_ use the injected header — "You need to contact the Gateway Ops team. File an ESECC (Enterprise Security External Control Council)".  We have not tried this route yet, nor did the old PW team, and it's described as a site-level redirect, rather than a page-level redirect.


## 2. [Redirect Implementation Strategy](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md)
`va.gov-team/platform/engineering/redirect-implementation-strategy.md`
_Dec 2022 update from me; prior update = Jan 2020_

4 types of redirects: 
1) Server-side redirects within [www.va.gov](http://www.va.gov/) ([link](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#1-server-side-redirects-within-wwwvagov))
2) Client-side redirects within [www.va.gov](http://www.va.gov/) ([link](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#2-client-side-redirects-within-wwwvagov))
3) Server-side redirects from subdomains (subdomain.va.gov) ([link](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#3-server-side-redirects-from-subdomains-subdomainvagov)) — used for full-site redirects
4) Client-side redirects for subdomains (subdomain.va.gov) ([link](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md#4-client-side-redirects-for-subdomains-subdomainvagov)) — used for page-level redirects. This is the same as 3b above, redirect via [vets-website/src/applications/proxy-rewrite/redirects/crossDomainRedirects.json](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/proxy-rewrite/redirects/crossDomainRedirects.json), which we tried and does not work for the redirect request in this ticket.


## 3. [Redirects Strategy Suggestion](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsp/teams/tools/frontend/redirects-strategy.md)
`va.gov-team/vsp/teams/tools/frontend/redirects-strategy.md`
_Feb 2019_

Doc #2, Redirect Implementation Strategy, specifically notes that it replaces this doc. 

## 4. [Redirects](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/information-architecture/request-redirect.md)
`va.gov-team/platform/information-architecture/request-redirect.md`
_April 2022_

IA doc specific to requesting a redirect & vetting it. 

## 5. [Client-side and service-side redirect implementation process for Public Websites team](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/public-websites/content-team-processes/URL-redirect-process)
`/va.gov-team/products/public-websites/content-team-processes/URL-redirect-process`
_June 2020_

Mostly an outdated workflow doc for an old PW team, that is replaced by doc #4 above, and should be updated or removed. 

## 6. [Client-side-redirects](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/public-websites/client-side-redirects)
`/va.gov-team/products/public-websites/client-side-redirects`
_Jan 2020_

Stub doc with no details, should be deleted.

## Other relevant docs
### [Reverse Proxy documentation](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/infrastructure/reverse_proxy.md)
`va.gov-team/platform/engineering/infrastructure/reverse_proxy.md`


# Docs to dos

- [ ] Consolidate [Redirects](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/teams/vsa/teams/public-websites/redirects.md)  info into  [Redirect Implementation Strategy](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md)
  - [ ] Remove contents of [Redirects](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/teams/vsa/teams/public-websites/redirects.md) and point to [Redirect Implementation Strategy](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md) 
- [ ] Mark  [Redirects Strategy Suggestion](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsp/teams/tools/frontend/redirects-strategy.md) clearly as deprecated in favor of [Redirect Implementation Strategy](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/redirect-implementation-strategy.md)
- [ ] Update title of the IA [Redirects](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/information-architecture/request-redirect.md) doc to be more clear about what it describes
- [ ] Mark [Client-side and service-side redirect implementation process for Public Websites team](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/public-websites/content-team-processes/URL-redirect-process) as deprecated or delete it altogether
- [ ] Delete [Client-side-redirects](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/public-websites/client-side-redirects)
- [ ] Update current [PW products doc Redirects section](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/public-websites#non-facility-redirects) with better pointers to final updated docs

**Comment by jilladams** [comment]
@apisandipas @ryguyk FYI: Latest change in PR [#23065](https://github.com/department-of-veterans-affairs/vets-website/pull/23065) is merged / deployed, and redirect is still not working.  Not urgent, as Dave has asked us to timebox this since it has sprawled. Maybe next week at onsite we can briefly chat at some point about how redirects and regroup.

**Comment by jilladams** [comment]
S76 sprint planning: the work to date has not successfully implemented a subdomain > subdomain redirect. 
* WebOps will not act on this, as they only handle full-site redirects.
* TeamSites leaves redirects of this kind to the site owner, which is Vet Center folks in this case.
* Vet Center site admins from TeamSite are not able to implement this redirect, and asked Public Websites to handle (via phone call with Jenny H-L & Jill).
* Our front-end client-side changes have not so far been successful. 

Per conversation with DaveC: we aren't able to spend more time on this right now. Low priority for this path, so we will not continue work here. We'll discuss next steps at the higher level subdomain > subdomain redirect level with DaveC and I'll close loop with the original requester, Tiffany, after that talk. Closing ticket.
